src/lib/selector/wallet.js

```diff
@@ -1,118 +1,155 @@
-import * as nearAPI from 'near-api-js';
-
-import BN from 'bn.js';
-
-const networks = {
-    mainnet: {
-        networkId: 'mainnet',
-        viewAccountId: 'near',
-        nodeUrl: 'https://rpc.mainnet.near.org',
-        walletUrl: 'https://wallet.near.org',
-        helperUrl: 'https://helper.mainnet.near.org'
-    },
-    testnet: {
-        networkId: 'testnet',
-        viewAccountId: 'testnet',
-        nodeUrl: 'https://rpc.testnet.near.org',
-        walletUrl: 'https://wallet.testnet.near.org',
-        helperUrl: 'https://helper.testnet.near.org'
-    },
-    localnet: {
-        networkId: 'localnet',
-        viewAccountId: 'test.near',
-    }
-}
+import * as nearAPI from "near-api-js";
 
-export class FastAuthWallet {
-    constructor({ signInContractId, networkId, ...rest }) {
-        this.networkId = networkId
-        this.signInContractId = signInContractId;
-        this.activeAccountId = window.localStorage.getItem('fast-auth:activeAccountId') || '';
-
-        this.keyStore = new nearAPI.keyStores.BrowserLocalStorageKeyStore();
-        this.near = new nearAPI.Near({
-            ...networks[networkId],
-            deps: { keyStore: this.keyStore },
-        });
-    }
+const {
+  transactions: { encodeSignedDelegate },
+} = nearAPI;
 
-    getContractId() {
-        return this.signInContractId;
-    }
+import BN from "bn.js";
 
-    getAccountId() {
-        return this.activeAccountId;
-    }
+import { networks } from "../../data/widgets";
 
-    async isSignedIn() {
-        return !!this.activeAccountId;
+export class FastAuthWallet {
+  constructor({ signInContractId, networkId, relayerUrl, ...rest }) {
+    this.networkId = networkId;
+    this.signInContractId = signInContractId;
+    this.activeAccountId =
+      window.localStorage.getItem("fast-auth:activeAccountId") || "";
+
+    this.keyStore = new nearAPI.keyStores.BrowserLocalStorageKeyStore();
+    this.near = new nearAPI.Near({
+      ...networks[networkId],
+      deps: { keyStore: this.keyStore },
+    });
+    this.relayerUrl = relayerUrl;
+  }
+
+  getContractId() {
+    return this.signInContractId;
+  }
+
+  getAccountId() {
+    return this.activeAccountId;
+  }
+
+  async isSignedIn() {
+    return !!this.activeAccountId;
+  }
+
+  async signIn() {
+    if (this.activeAccountId) return;
+
+    const accountCreationData = JSON.parse(
+      window.localStorage.getItem("fast-auth:account-creation-data") ||
+        JSON.stringify({})
+    );
+
+    try {
+      if (
+        !accountCreationData.limitedAccessKey ||
+        !accountCreationData.isCreated
```

Comment:
I might be missing something here, but shouldn't this be:
```suggestion
        accountCreationData.isCreated
```

Reply:
This case is a bit counter-intuitive -- after creating the account via recovery service, we redirect to the app after setting a flag indicating that the account was created successfully -- so this is correct, if the account wasn't created we can't yet fire the signIn() logic here